### PR TITLE
add sort function to sort employees 

### LINKED ISSUE #47: As a user I can sort employee records by name
… so that I can view employee records in alphabetical order

### LINKED ISSUE #48: As a user I can sort employee records by date of joining
… so that I can find the employees who stayed the shortest/longest in the company

### LINKED ISSUE #49: As a user I can sort employee records by salary
… so that I can view the salary structure in my company

### LINKED ISSUE #50: As a user I can sort employee records by date of birth
… so that I can find the youngest/oldest employees in the company

### PR DESCRIPTION
**Pull request by daiweinus**
close #47, close #48, close #49, close #50  add employee sort function by name, salary, date of join & date of birth

### CODE REVIEW

File: src/main/java/employeetracker/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package employeetracker.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import employeetracker.model.Model;
+
+
+/**
+ * Sort and lists all persons in address book by name contains any of the argument keywords.
```
Comment (wz-chan) [inline src/main/java/employeetracker/logic/commands/SortCommand.java]: Perhaps this should be `list all employees in Employee Tracker` instead to make it more consistent with our naming conventions?

File: src/main/java/employeetracker/model/EmployeeTracker.java [hunk 2]
```diff
@@ -86,6 +86,14 @@ public void setEmployee(Employee target, Employee editedEmployee) {
         employees.setEmployee(target, editedEmployee);
     }
 
+    /**
+     * Removes {@code key} from this {@code EmployeeTracker}.
+     * {@code key} must exist in the address book.
```
Comment (wz-chan) [inline src/main/java/employeetracker/model/EmployeeTracker.java]: Perhaps `Employee Tracker` is more consistent with our naming convention than `address book`?

**Review by wz-chan — CHANGES_REQUESTED** [review]
Your code is definitely good to merge except for some minor issues which I've highlighted

File: src/main/java/employeetracker/logic/commands/SortCommand.java (hunk 1 above)
Comment (daiweinus) [reply src/main/java/employeetracker/logic/commands/SortCommand.java]: done

**Review by daiweinus — COMMENTED** [review]
comment review

**Review by wz-chan — APPROVED** [review]
LTGM

**Pull request merged**